blog/_posts/2017-09-04-gsoc-NeuralNetDiffEq.md

```diff
@@ -0,0 +1,191 @@
+---
+layout: post
+title: "NeuralNetDiffEq.jl: A Neural Network solver for Ordinary Differential Equations."
+author: Akshay Sharma
+---
+
+My GSoC 2007 [project](https://summerofcode.withgoogle.com/projects/#5850956641075200) was to implement a package for Julia to solve Ordinary Differential Equations using Neural Networks.
+The purpose of the project was to provide an additional DE solver using Neural Networks which has parallelism in time as the key advantage over other solvers which are iterative in nature. The project was based on a [research paper](https://arxiv.org/pdf/physics/9705023.pdf) which proposed the function approximation capabilities of Neural Networks for solving differential equations. The project was a mixture of research as well as implementation aspects and still has a few parts left to work upon.
+I chose to work on this project as I have interest in Mathematics and Machine Learning and it involved concepts of both the fields. The package uses [DifferentialEquations.jl](https://github.com/JuliaDiffEq/DifferentialEquations.jl) for the solver interface and [KNet.jl](https://github.com/denizyuret/Knet.jl) for NN solver implementation.
+
+## How to use Neural Network for solving Differential Equations?
+The concept of this solver is based on the UAT (Universal Approximation Theorem) which says that a Neural Network with not more than 3 hidden can approximate any continuous function. The Neural Network is made to minimize a loss function which then results in the convergence of our Trial solution towards the actual (analytical) solution. To know more about UAT [click here](http://neuralnetworksanddeeplearning.com/chap4.html).
+
+## Research aspect of the project and The Challenge
+The research paper we referred on the topic is quite old and lacks proper explanation as well as examples. The research done on this part of Neural Networks is not too much and we were not able to get much help from the research papers related to the topic.
```

Comment:
Better might be to say that understanding the examples and explanations from the paper was challenging.
It actually had a fair few examples, but they were not easy to handle